app/models/post_action_type.rb

```diff
@@ -79,30 +79,24 @@ def is_flag?(sym)
       flag_types.valid?(sym)
     end
 
-    private
-
     def initialize_flag_settings
+      @types = nil
       @flag_settings = FlagSettings.new
-      @flag_settings.add(3, :off_topic, notify_type: true, auto_action_type: true)
-      @flag_settings.add(
-        4,
-        :inappropriate,
-        topic_type: true,
-        notify_type: true,
-        auto_action_type: true,
-      )
-      @flag_settings.add(8, :spam, topic_type: true, notify_type: true, auto_action_type: true)
-      @flag_settings.add(6, :notify_user, topic_type: false, notify_type: false, custom_type: true)
-      @flag_settings.add(
-        7,
-        :notify_moderators,
-        topic_type: true,
-        notify_type: true,
-        custom_type: true,
-      )
-      @flag_settings.add(10, :illegal, topic_type: true, notify_type: true, custom_type: true)
-      # When adding a new ID here, check that it doesn't clash with any added in
-      # `ReviewableScore.types`. You can thank me later.
+      if ActiveRecord::Base.connection.table_exists?(:post_flags)
+        PostFlag
+          .enabled
+          .order(:position)
+          .find_each do |post_flag|
```

Comment:
I don't think `find_each` is needed here, there won't be so many that this needs to be batched up. `.each` should be fine